app/frontend/components/pages/dashboard/dashboardPage.tsx

```diff
@@ -22,202 +21,249 @@ import AccountsDashboard from '../accounts/accountsDashboard'
 import {State} from '../../../state'
 import PoolOwner from '../advanced/poolOwner'
 import ErrorModals from './errorModals'
+import {useEffect, useState} from 'preact/hooks'
+import {SubTabs, MainTabs} from '../../../constants'
 
-interface Props {
-  selectedMainTab: any
-  selectMainTab?: (value: string) => void
+// we could export this and use it in other places, in that case
+// we should optimalize it with context
+// https://blog.logrocket.com/developing-responsive-layouts-with-react-hooks/
+const useViewport = () => {
+  const [isViewportSmall, setIsViewportSmall] = useState(false)
+
+  const checkScreenSize = () => {
+    setIsViewportSmall(window.innerWidth < 1024)
+  }
+  useEffect(() => {
+    checkScreenSize()
+    window.addEventListener('resize', checkScreenSize)
+
+    return () => window.removeEventListener('resize', checkScreenSize)
+  }, [])
+
+  return isViewportSmall
 }
 
-const StakingPage = () => {
+const StakingPage = ({isViewportSmall}) => {
+  const subTabs = [SubTabs.DELEGATE_ADA, SubTabs.CURRENT_DELEGATION, SubTabs.STAKING_HISTORY]
+  const defaultSubTab = SubTabs.DELEGATE_ADA
+  const mainSubTab = SubTabs.SHELLEY_BALANCES
   return (
-    <div className="dashboard desktop">
-      <div className="dashboard-column">
-        <ShelleyBalances />
-        <StakingHistoryPage />
-      </div>
-      <div className="dashboard-column">
-        <DelegatePage />
-        <CurrentDelegationPage />
-      </div>
-    </div>
+    <Fragment>
+      {isViewportSmall ? (
+        <div className="dashboard mobile">
+          <DashboardMobileContent
+            subTabs={subTabs}
+            defaultSubTab={defaultSubTab}
+            mainSubTab={mainSubTab}
+          />
+        </div>
+      ) : (
+        <div className="dashboard desktop">
+          <div className="dashboard-column">
+            <ShelleyBalances />
+            <StakingHistoryPage />
+          </div>
+          <div className="dashboard-column">
+            <DelegatePage />
+            <CurrentDelegationPage />
+          </div>
+        </div>
+      )}
+    </Fragment>
   )
 }
 
-const SendingPage = ({shouldShowExportOption}) => {
+const SendingPage = connect((state: State) => ({
+  shouldShowExportOption: state.shouldShowExportOption,
+}))(
```

Comment:
I'd perhaps not create a HOC out of this and have 'shouldShowExportOption' passed as a prop from higher, what do u think? Also, this component has the "isViewportSmall: boolean" prop typed, others do not

Reply:
I didn't want to connect the dashboard page to `shouldShowExportOption` just so it passes it elsewhere, but doesn't matter, fixed, and typed.